node/network/gossip-support/src/tests.rs

```diff
@@ -475,6 +475,64 @@ fn issues_connection_request_to_past_present_future() {
 	});
 }
 
+#[test]
+fn disconnect_when_not_in_past_present_future() {
+	sp_tracing::try_init_simple();
+	let hash = Hash::repeat_byte(0xAA);
+	test_harness(make_subsystem(), |mut virtual_overseer| async move {
+		let overseer = &mut virtual_overseer;
+		overseer_signal_active_leaves(overseer, hash).await;
+		assert_matches!(
+			overseer_recv(overseer).await,
+			AllMessages::RuntimeApi(RuntimeApiMessage::Request(
+				relay_parent,
+				RuntimeApiRequest::SessionIndexForChild(tx),
+			)) => {
+				assert_eq!(relay_parent, hash);
+				tx.send(Ok(1)).unwrap();
+			}
+		);
+
+		assert_matches!(
+			overseer_recv(overseer).await,
+			AllMessages::RuntimeApi(RuntimeApiMessage::Request(
+				relay_parent,
+				RuntimeApiRequest::SessionInfo(s, tx),
+			)) => {
+				assert_eq!(relay_parent, hash);
+				assert_eq!(s, 1);
+				let mut heute_leider_nicht = make_session_info();
```

Comment:
Would be great to translate this 😜

Reply:
That's an easter egg :stuck_out_tongue:

Reply:
LOL - I totally missed that one. :see_no_evil: